### PR TITLE
fix v2 procedure

### PR DESCRIPTION
**Pull request by GongChangYan**
1.  Fixed v2 procedure issue. Snapshot can be created from a single txn in Olap. Added six algorithm plugin v2 versions of bfs, cdlp, wcc, lcc, pagerank, and sssp
2. Supports uploading multiple cpp files when uploading plugins

### CODE REVIEW

File: src/plugin/plugin_manager.cpp [hunk 1]
```diff
@@ -451,17 +467,29 @@ bool lgraph::SingleLanguagePluginManager::LoadPluginFromCode(
...
+    std::string all_codes;
+    if (code_type == plugin::CodeType::CPP) {
+        all_codes = MergeCodeFiles(code, filename, name);
+    }
     switch (code_type) {
     case plugin::CodeType::SO:
         break;
     case plugin::CodeType::PY:
-        exe = CompilePluginFromCython(name, code);
+        if (code.size() != 1) {
```
[the remark this answers is not in the document]
Comment (GongChangYan) [reply src/plugin/plugin_manager.cpp:478]: py没这个需求

File: src/client/cpp/restful/rest_client.cpp [hunk 2]
```diff
@@ -1051,8 +1051,17 @@ bool RestClient::LoadPlugin(const std::string& db, lgraph_api::PluginCodeType ty
     body[RestStrings::READONLY] = json::value::boolean(plugin_info.read_only);
```
[the remark this answers is not in the document]
Comment (GongChangYan) [reply src/client/cpp/restful/rest_client.cpp:1051]: tugraph-web也只能上传一个。tugraph-web只是改了code_base64的数据类型，从string改成string[]，是为了和后端的参数对应上，实际前端选的时候只能选一个。
restclient应该是写漏了，我加上

File: src/cypher/procedure/procedure.cpp [hunk 3]
```diff
@@ -1678,7 +1678,8 @@ void BuiltinProcedure::DbPluginLoadPlugin(RTContext *ctx, const Record *record,
     fma_common::encrypt::Base64 base64;
     std::string content = base64.Decode(args[2].String());
     bool success =
-        db.LoadPlugin(plugin_type_it->second, ctx->user_, args[1].String(), content,
+        db.LoadPlugin(plugin_type_it->second, ctx->user_, args[1].String(),
```
[the remark this answers is not in the document]
Comment (GongChangYan) [reply src/cypher/procedure/procedure.cpp:1681]: 不太好做

File: test/test_procedures/custom_shortestpath.cpp
[the remark this answers is not in the document]
Comment (GongChangYan) [reply test/test_procedures/custom_shortestpath.cpp:1]: 都改成v2了

**Review by qishipengqsp — APPROVED** [review]

**Review by lipanpan03 — APPROVED** [review]

**Pull request merged**